xdsl/dialects/smt.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import TypeAlias
+
+from xdsl.dialects.builtin import BoolAttr, IntegerAttr
+from xdsl.ir import Dialect, ParametrizedAttribute, SSAValue, TypeAttribute
+from xdsl.irdl import (
+    AtLeast,
+    IRDLOperation,
+    RangeOf,
+    base,
+    irdl_attr_definition,
+    irdl_op_definition,
+    prop_def,
+    result_def,
+    traits_def,
+    var_operand_def,
+)
+from xdsl.traits import ConstantLike, Pure
+
+
+@irdl_attr_definition
+class BoolType(ParametrizedAttribute, TypeAttribute):
+    """A boolean."""
+
+    name = "smt.bool"
+
+
+NonFuncSMTType: TypeAlias = BoolType
```

Comment:
Is this used anywhere?

Reply:
It will in the next commit, I can remove it now though